src/net/ossl.cc

```diff
@@ -816,8 +854,30 @@ class session : public enable_shared_from_this<session>, public session_impl {
 
 
 private:
+    std::shared_ptr<X509> get_peer_certificate() const {
```

Comment:
Reason i did this is because this method either returns what i've cached or what is returned by SSL_get_peer_cert. Cached i have a unique_ptr, so i can only return a const ref to that.